### PR TITLE
Add/nipt/QC check of flow cell before upload & latest sequenced flow cell usage

### PR DESCRIPTION
**Pull request by karlnyr**
### Added

- Added QC check of flowcell before upload and latest sequenced flowcell usage

### How to prepare for tests

- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`
- [x] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh feat/fluffy-flowcell-improvements`

### How to test

- [x] Find case with two flowcells connected to it.
- [x] cg workflow fluffy link <nipt_case_id>
- [x] cg upload nipt statina case <nipt_case_id>

### Expected test outcome

- [x] Only files from the latest flowcell are linked
- [x] Batch is uploaded to statina
- [x] Take a screenshot and attach or copy/paste the output.

## Review

- [x] code approved by @Mropat 
- [x] tests executed by @karlnyr 
- [x] "Merge and deploy" approved by @Mropat 
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [x] Document in [nipt-analysis](https://github.com/Clinical-Genomics/atlas/pull/401)
- [x] Deploy this branch 
- [x] Inform to @Clinical-Genomics/production

### CODE REVIEW

**Comment by karlnyr** [comment]
Flow cells related to samples:
image


Testing linking:
image

Fetched files with fastq AND flowcell tag

Testing upload:
image

Case is uploaded to statina without issues(did not do for all db's due to stage/prod not being separate dps for the FTP)

File: cg/meta/workflow/fluffy.py [hunk 1]
```diff
@@ -92,14 +92,17 @@ def link_fastq_files(self, case_id: str, dry_run: bool = False) -> None:
...
         workdir_path = self.get_workdir_path(case_id=case_id)
         if workdir_path.exists() and not dry_run:
             LOG.info("Fastq directory exists, removing and re-linking files!")
             shutil.rmtree(workdir_path, ignore_errors=True)
         workdir_path.mkdir(parents=True, exist_ok=True)
         for family_sample in case_obj.links:
             sample_id = family_sample.sample.internal_id
-            files: Query = self.housekeeper_api.files(bundle=sample_id, tags=["fastq"])
+            files: Query = self.housekeeper_api.files(
+                bundle=sample_id, tags=["fastq", latest_flow_cell.name]
```
Comment (Mropat) [inline cg/meta/workflow/fluffy.py:104]: this will break eventually, for example if we try to re-deliver old fastq what lost the tag due to spring

**Comment by Mropat** [comment]
you can also add dry-run for SFTP upload method, so that both STFP and statina upload are stopped/initiated with one command

**Review by Mropat — APPROVED** [review]

File: cg/meta/workflow/fluffy.py (hunk 1 above)
Comment (karlnyr) [reply cg/meta/workflow/fluffy.py:104]: will adress the fact that fastqs are not stored with fc-tag after decompression in another PR

**Comment by karlnyr** [comment]
iInstalled in prod:
image

**Pull request merged**